quic/s2n-quic-transport/src/ack/ack_manager.rs

```diff
@@ -212,7 +222,31 @@ impl AckManager {
         //
         // Most likely, this packet is very old and the contents have already
         // been retransmitted by the peer.
-        if self.ack_ranges.insert_packet_number(packet_number).is_err() {
+        if let Err(err) = self.ack_ranges.insert_packet_number(packet_number) {
+            match err {
+                AckRangesError::RangeInsertionFailed { min, max }
+                | AckRangesError::LowestRangeDropped { min, max } => {
+                    let start = self
+                        .ack_ranges
+                        .min_value()
+                        .expect("should be non empty")
+                        .into_event();
+                    let end = self
+                        .ack_ranges
+                        .max_value()
+                        .expect("should be non empty")
+                        .into_event();
+
+                    publisher.on_ack_processed(AckProcessed {
+                        action: AckAction::RxAckRangeDropped {
+                            packet_number_range: min.into_event()..max.into_event(),
+                            capacity: self.ack_ranges.intervals().count() as u16,
```

Comment:
`count` iterates over each one of the intervals, which isn't needed when you can just get the `interval_len` directly.

Reply:
really good catch! didnt realize count was on the iterator and it iterates over the items!